aspnet/fundamentals/logging.rst

```diff
@@ -3,117 +3,171 @@
+The remainder of this article explains some details and options for logging:
 
-Error
-  An error should be logged when the current flow of the application must stop due to some failure, such as an exception that cannot be handled or recovered from. These messages should indicate a failure in the current activity or operation (such as the current HTTP request), not an application-wide failure. Example: ``Cannot insert record due to duplicate key violation``
+* `Log category`_
+* `Log level`_
+* `Log event ID`_
+* `Log message format string`_
+* `Logging exceptions`_
+* `Log filtering`_
+* `Log scopes`_
+* `TraceSource logging`_
+* `Third-party logging frameworks`_
 
-Critical
-  A critical log level should be reserved for unrecoverable application or system crashes, or catastrophic failure that requires immediate attention. Examples: data loss scenarios, out of disk space
+Log category
+------------
 
-The ``Logging`` package provides `helper extension methods <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/Extensions/Logging/LoggerExtensions/index.html>`_ for each ``LogLevel`` value, allowing you to call, for example, ``LogInformation``, rather than the more verbose ``Log(LogLevel.Information, ...)`` method. Each of the ``LogLevel``-specific extension methods has several overloads, allowing you to pass in some or all of the following parameters:
+Included with each log is a field that is named *category*. The category may be any string but is often the fully qualified name of the class from which the logs are written.  For example: "TodoApi.Controllers.TodoController".
 
-string data
-  The message to log.
+You specify the category when you create a logger object or request one from DI, and the category is automatically included with every log written by that logger. You can specify the category explicitly or you can use an extension method that derives the category from the type. To specify the category explicitly, request an `ILoggerFactory` object from DI and pass in the category to the ``CreateLogger`` method, as shown below.
 
-EventId eventId
-  A numeric id to associate with the log, which can be used to associate a series of logged events with one another. Event IDs should be static and specific to a particular kind of event that is being logged. For instance, you might associate adding an item to a shopping cart as event id 1000 and completing a purchase as event id 1001. This allows intelligent filtering and processing of log statements.
+.. literalinclude:: logging/sample/src/TodoApi/Controllers/TodoController.cs
+  :language: c#
+  :start-after: snippet_CreateLogger
+  :end-before: endregion
+  :emphasize-lines: 7,10
 
-string format
-  A format string for the log message.
+As a shortcut, you can request from DI an instance of ``ILogger<T>``, as in the following example. This is equivalent to requesting an ``ILoggerFactory`` and calling ``CreateLogger`` with the fully qualified type name of ``T``.
```

Comment:
Reworded but kept the present order of the two as I think it's clearer that way.